pkg/operator/k8sutil/cmdreporter/cmdreporter.go

```diff
@@ -144,8 +144,7 @@ func (cr *CmdReporter) Job() *batch.Job {
 // and retcode of the command as long as the image ran it, even if the retcode is nonzero (failure).
 // An error is reported only if the command was not run to completion successfully. When this
 // returns, the ConfigMap is cleaned up (destroyed).
-func (cr *CmdReporter) Run(timeout time.Duration) (stdout, stderr string, retcode int, retErr error) {
-	ctx := context.TODO()
+func (cr *CmdReporter) Run(ctx context.Context, timeout time.Duration) (stdout, stderr string, retcode int, retErr error) {
```

Comment:
I've added `ctx context.Context` to Run function. It seems we can merge `ctx` and `timeout` by using `context.WithTimeout`. What do you think?

Reply:
Yes definitely, do you want to rework `waitForConfigMap` in this PR?  Since `cr.newWatcher()` now accepts a context it should be straightforward. Thanks!

Reply:
I've updated the PR. Now API calls in `cr.newWatcher()` handle cancellation but also timeout.